app/controllers/products_controller.rb

```diff
@@ -1,10 +1,42 @@
 class ProductsController < ApplicationController
   def index
+    @products = Product.all
   end
 
   def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.create(product_params)
+    if @product.save
+      redirect_to products_path
+    else
+      render :new
+    end
   end
 
   def edit
+    @product = Product.find(params[:id])
+  end
+
+  def update
+    @product = Product.find(params[:id])
+    if @product.update(product_params)
+      redirect_to products_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @product = Product.find(params[:id])
```

Comment:
@amitshethe Product.find is repeated in edit, update & destroy.
You can use controller filters to avoid repeatition.